src/CommunityToolkit.Maui.Markup.SourceGenerators/SourceGenerators/TextAlignmentExtensionsGenerator.cs

```diff
@@ -1,377 +1,368 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Collections.Immutable;
-using System.Linq;
-using System.Text;
-using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.CSharp;
-using Microsoft.CodeAnalysis.CSharp.Syntax;
-using Microsoft.CodeAnalysis.Text;
-
-namespace CommunityToolkit.Maui.Markup.SourceGenerators;
-
-[Generator(LanguageNames.CSharp)]
-class TextAlignmentExtensionsGenerator : IIncrementalGenerator
-{
-	const string iTextAlignmentInterface = "Microsoft.Maui.ITextAlignment";
-	const string mauiControlsAssembly = "Microsoft.Maui.Controls";
-
-	public void Initialize(IncrementalGeneratorInitializationContext context)
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace CommunityToolkit.Maui.Markup.SourceGenerators;
+
+[Generator(LanguageNames.CSharp)]
+class TextAlignmentExtensionsGenerator : IIncrementalGenerator
+{
+	const string iTextAlignmentInterface = "Microsoft.Maui.ITextAlignment";
+	const string mauiControlsAssembly = "Microsoft.Maui.Controls";
+
+	public void Initialize(IncrementalGeneratorInitializationContext context)
 	{
 		// Get All Classes in User Library
-		var userGeneratedClassesProvider = context.SyntaxProvider.CreateSyntaxProvider(
-			static (syntaxNode, cancellationToken) => syntaxNode is ClassDeclarationSyntax { BaseList: not null },
-			static (context, cancellationToken) =>
-			{
-				var compilation = context.SemanticModel.Compilation;
-
-				var iTextAlignmentInterfaceSymbol = compilation.GetTypeByMetadataName(iTextAlignmentInterface);
-
-				if (iTextAlignmentInterfaceSymbol is null)
-				{
-					throw new Exception("There's no .NET MAUI referenced in the project.");
-				}
-
-				var classSymbol = (INamedTypeSymbol?)context.SemanticModel.GetDeclaredSymbol(context.Node);
-
-				// If the ClassDlecarationSyntax doesn't implements those interfaces we just return null
-				if (classSymbol?.AllInterfaces.Contains(iTextAlignmentInterfaceSymbol, SymbolEqualityComparer.Default) is not true)
-				{
-					return null;
-				}
-
-				return classSymbol;
-			});
+		var userGeneratedClassesProvider = context.SyntaxProvider.CreateSyntaxProvider(
+			static (syntaxNode, cancellationToken) => syntaxNode is ClassDeclarationSyntax { BaseList: not null },

```

Comment:
Note: this will produce multiple matches if someone has a type split into multiple partial declarations, and there's some base type declared in each partial declaration. What I do in the MVVM Toolkit is I only select the tree that matches the first metadata reference for the symbol. There's an extension you can borrow from there if you need, otherwise just ignore 🙂

Reply:
Yeah good point.